### ISSUE TITLE
Forked it, tossed it in docker hub.....

### ISSUE THREAD

**Issue by faddat** [spujadas/flarum-docker#1]
It is blowing up as follows on Docker Hub:

The command [/bin/sh -c mkdir -p ${FLARUM_ROOT}/flarum/core     && cd ${FLARUM_ROOT}/flarum/core    && git clone https://github.com/flarum/core .   && mkdir public] returned a non-zero code: 1

And on my PC, it is blowing up as follows:
Service 'flarum' failed to build: The command [/bin/sh -c apt-get install -qqy curl unzip git-core ack-grep software-properties-common build-essential  php5-cli php5-fpm php5-mysql php5-pgsql php5-sqlite php5-curl php5-gd php5-gmp php5-mcrypt php5-memcached php5-imagick php5-intl php5-xdebug] returned a non-zero code: 100

When I get it working, I'll toss out a pull request.  Thanks for maknig it!

**Comment by spujadas** [comment]
Thanks for the feedback.

I haven't rebuilt the Flarum image recently, but apparently the structure of the Flarum source was completely revamped over the past few weeks (including making flarum/core a submodule of flarum/flarum), so forking+building as is (git cloning the latest source along the way) is likely to fail at some point. I'll take a look and might update the "git clone ... && …" command to check out a specific commit of the source to avoid this in future in case the structure changes again.

As for that apt-get command failing on your PC with a generic 100 error code, that _is_ strange... could be anything. :confused: 

Anyway give me a few days to update the repo, at least to get it aligned with the latest Flarum tree, and we'll see how it goes (including testing if you can get it working on your PC).

**Comment by faddat** [comment]
oh, I'm happy to update it with you in fact.  I need to learn how to containerize laravel, anyway.... if I can figure it out, I'll do it right now :).

**Comment by spujadas** [comment]
Hey, sure thing!
If it can help [here's an updated Dockerfile](https://gist.github.com/spujadas/119da7424dafc28fa5db) that I started to work on to reflect the new structure of Flarum.

On my PC it works up to line 210 (`php artisan flarum:install`) and then the `artisan` command itself no longer works (crashes with `[ErrorException] Invalid argument supplied for foreach()`), which is weird to say the least.
I started looking into this earlier today, haven't sorted it out yet, but I guess that it has something to do with a `config.php` file getting created at https://github.com/flarum/core/blob/825b4082defbcaa53bfce3d4d284ec317a9e014e/src/Console/InstallCommand.php#L52 (deleting the file makes the `artisan` command work again).

Oh, and the `start.sh` script needs to be updated as it currently runs `ember serve` but Flarum no longer uses Ember.js.

Anyway, you're quite welcome to give it a shot!
